modules/partition-distribution/src/test/java/org/apache/ignite/internal/partitiondistribution/AssignmentsSerializerTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.partitiondistribution;
+
+import static java.util.Comparator.comparing;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.Set;
+import org.apache.ignite.internal.versioned.VersionedSerialization;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class AssignmentsSerializerTest {
+    private final AssignmentsSerializer serializer = new AssignmentsSerializer();
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void serializationAndDeserialization(boolean force) {
+        Set<Assignment> nodes = Set.of(Assignment.forPeer("a"), Assignment.forLearner("b"));
+        Assignments originalAssignments = force ? Assignments.forced(nodes, 1000L) : Assignments.of(nodes, 1000L);
+
+        byte[] bytes = VersionedSerialization.toBytes(originalAssignments, serializer);
+        Assignments restoredAssignments = VersionedSerialization.fromBytes(bytes, serializer);
```

Comment:
lets also check that serialization of restoredAssignments also produces those hardcoded bytes, if it's important.

Reply:
The test is about ability to restore (in the future) something serialized with version 1 (in the past). The check you mention is not needed in this scenario. Even more, if we change the format and start serializing with version 2, this check will start failing.